### DISCUSSION TITLE
Move around parentheses

### DISCUSSION THREAD

**Discussion by cloudhan** [helix-editor/helix#7989] [Q&A] [answered]
In vim `%` can be used to move the cursor in enclosing pairs of brackets

```
(123 + 456 * (789 - 000) / nan)
0            1         1      0
```

place cursor at any position above 0 or 1, press `%` can quickly jump back and forth. Is there an equivalent?

And, is there is way to quickly jump outside of it, say, at `)` marked as `1`, jump to `)` marked as `0`

**Comment by CptPotato** [comment, marked as answer]
There's `mm` if this is what you're looking for. (see [docs](https://docs.helix-editor.com/keymap.html#match-mode))

If you want to jump forward to a closing parenthesis (or any other char) you can use `f)`.

**Reply by CptPotato** [reply]
> And, is there is way to quickly jump outside of it, say, at `)` marked as `1`, jump to `)` marked as `0`

Forgot to mention, there's also a tabout feature (#4443, not in the current release yet) that lets you jump up the syntax tree using tab in insert mode. It will not exactly jump from on parenthesis to the next as you outlined but you might find it useful regardless.
